### ISSUE TITLE
Cannot Log in to Lab VM´s

### ISSUE THREAD

**Issue by Frank-Rei** [microsoft/MSLab#295]
After the first Scenario was build on my Testing Machine i cannot Log in to any VM in any way. Error Messages are eiter: "The logon method used is not allowed" or "For remote logon, you must be authorized to log on through Remote Desktop Services. The members of the Administrators group are equipped with this right by default."
I tried MSDN Iso´s en_windows_server_2016_updated_feb_2018_x64_dvd_11636692.iso and de_windows_server_2019_x64_dvd_17559a5b.iso for creating Parent Disks. I also tried providding the LabConfig.ps1 with another Adminname and Password but always got the same Errors. Maybe i made a configuration Mistake but i have no idea what it is.

**Comment by jaromirk** [comment]
When was adminname and password modified? Before 2_CreateParentDisks.ps1 ? or just before deploy. In step 2 it's hardcoded to DC... in step 3 it just uses these creds to connect to dc to djoin machines (generate djoin blobs)

**Comment by Frank-Rei** [comment]
Hi Jaromir. I was aware of that and therefore i deleted all Folders created by Steps 1 and 2 before i modified the LabConfig,ps1 to use a different Adminname and Password. After that is started again with  Step 1, Step 2 and Strep 3. Again, the Credentials were valid but had no Permission to log in to any VM.

**Comment by jaromirk** [comment]
hmm, can you share deploy log? transcript text file that's created during lab deploy... you are also not able to log in into DC? if you are able, can you check if lab vms are in AD?

**Comment by Frank-Rei** [comment]
No, i am not able to Log in into the DC VM. I will try it again tomorrow with an unmodified LabConfig as soon as i´m in the Office and ulpload the Logs created by the Installer Scrips if it fails another time.

**Comment by jaromirk** [comment]
well, it's weird. it means DC was not properly built. can you send me labconfig to jaromirk@microsoft.com so I can try a repro?

**Comment by Frank-Rei** [comment]
I now have tried it once more. I have downloaded all necesary Files and left the LabConfig unmodified. Now everything seems to work as intended. I was able to log in into the DC anc S2D1 VM´s without Problems (first Time i ran into that issue that the S2Dx Servers were not Domain joined when all startet at once). 

Whats different now?
- i used the english Version of Server 2019 MSDN instead of german.
- i left the LabConfig.ps1 untouched. 
My modifications were Adminname+Password, InstallSCVMM='Yes' and ParentVHD = 'Win2016_G2.vhdx' (i used Server 2016 iso for my second try)

No i have no automated SCVMM installation but i´m fine with that. It basically works and thats all that matters to me. If you still want my modified modified labconfig for further investigation i could of course email it to you. 

At this Point i´d like to thank you very much for your quick response and this Project in General. Your work is highly appreciated.

**Comment by jaromirk** [comment]
Cool! For InstallSCVMM="Yes" just add install files into tools disk (there are place holder for it) and it will be automagically installed. AdminName should be something else than Administrator (as Administrator is built in already).

**Comment by Frank-Rei** [comment]
Yes, i think i have put all necessary files in the correct locations because the Script processed them without errors. But without being able to log in i could not verify that :).

**Comment by jaromirk** [comment]
OK, send labconfig over - I'll take a look. :)

**Comment by Frank-Rei** [comment]
You got Mail :)

**Comment by jaromirk** [comment]
hmm, did not arrive. make sure it's either online, or txt (.ps1 is filtered out)

**Comment by Frank-Rei** [comment]
I sent it to you as an zipped txt file. Hope this gets through.

**Comment by jaromirk** [comment]
OK, so I validated your labconfig and it works just fine (with Windows Server 2019 and EN ISO) 
![image](https://user-images.githubusercontent.com/9829838/70044802-4e055600-15c3-11ea-9189-41412ad17fea.png)

**Comment by razza-guhl** [comment]
Hello,

i had the same issue with german Windows Server 2019 (VLSC, 1809.13_64Bit_German) iso file. Cannot login ether with basic or extend session through Hyper-V console, as @Frank-Rei stated.

Recreated everything from scratch with english ISO version fixed the login problem.

Thank you for the scripts!

**Comment by jaromirk** [comment]
Hi Razza, sorry, I guess it's because different administrator name. DC itself works? Or the created VMs have an issue?

**Comment by razza-guhl** [comment]
The DC had the issue. I didnt change the administrator username or password, just left the default settings as is (LabAdmin, LS!setup!).

After trying to login to DC I had the same error messages as OP.
"The logon method used is not allowed" or "For remote logon, you must be authorized to log on through Remote Desktop Services. The members of the Administrators group are equipped with this right by default." - but in german language as I used german ISO file.

Do you need any logs?

**Comment by ourIThome** [comment]
Might not be anything, but try typing the password in the username field.  I know s and z are different in keyboard as well as some special symbols.  Add I say, probably nothing ..

**Comment by jaromirk** [comment]
I think it will be an issue with "administrator" name in German. Is it different from "Administrator"?

In MSLab it's hardcoded :/

https://github.com/microsoft/MSLab/blob/master/Scripts/2_CreateParentDisks.ps1#L626

**Comment by razza-guhl** [comment]
The user "Administrator" works with the default password "LS1setup!". So i could take a look whats inside the AD after the deployment.

It seems to be an language issue/mismatch with the builtin group names in AD. 

- English: "Domain Admins" => German: "Domänen-Admins" 
- English: "Enterprise Admins" => German: "Organisations-Admins"
- English: "Schema Admins" => German: "Schema-Admins"

The script tries to assign the DomainAdminUser "LabAdmin" to those AD groups. 

Here is an example of group assignment for "Domain Admins" in the script:
https://github.com/microsoft/MSLab/blob/master/Scripts/2_CreateParentDisks.ps1#L783

As "Domain Admins" and the other groups does not exists in german AD, the groups are created by the script. These groups are created as common security groups without a "role" within the (german) AD. 
This results in the DomainAdminUser "LabAdmin" is actually just a domain user.

**Comment by jaromirk** [comment]
Hmm, problem is, that AD DSC accepts names only.

https://github.com/dsccommunity/ActiveDirectoryDsc/wiki/ADGroup

So for German language (and any other language we would need to add to MSLab, we would need to create "translation table").

I guess easiest "workaround" would be to use English ISO and then generate another German VHD to parent disk and use that one.

**Comment by ourIThome** [comment]
Can well known SIDS help... https://docs.microsoft.com/en-us/troubleshoot/windows-server/identity/security-identifiers-in-windows#well-known-sids-all-versions-of-windows

**Comment by jaromirk** [comment]
Not in DSC. We would need to run custom script instead. DSC module accepts names only (well I did not try ID, but documentation says name)...
